cranelift/codegen/src/isa/riscv64/lower/isle.rs

```diff
@@ -82,6 +82,9 @@ impl generated_code::Context for RV64IsleContext<'_, '_, MInst, Riscv64Backend>
             self.lower_ctx.sigs(),
             callee_sig,
             &callee,
+            // TODO: this should be Opcode::ReturnCall, once riscv64 has been ported to the new
+            // tail call strategy.
+            Opcode::Call,
```

Comment:
Is this going to break return calls on non-x64 until we get around to poring these architectures to the new tail call strategy? I don't think this PR should regress any existing functionality.

Reply:
Nope, we're using `Call` instead of `ReturnCall` here specifically to avoid breaking aarch64 and riscv64 until we can get around to applying similar changes to them. `from_func` was always using `Call` for the opcode before so passing that here now doesn't change what data structures that function constructs, and then we rely on the distinction in the new `is_tail_call` helper to ensure that the generated code isn't changed.